nailgun/entities.py

```diff
@@ -5244,6 +5244,12 @@ def read(self, entity=None, attrs=None, ignore=None):
         if ignore is None:
             ignore = set()
         ignore.add('organization')
+        if attrs is None:
+            # Workaround for
+            # https://github.com/SatelliteQE/nailgun/issues/475
+            attrs = self.read_json()
+            for system in attrs.get('systems', []):
+                system['id'] = system['host_id']
```

Comment:
Think preferably to verify that id is not in system and that host_id is there to allow this to work for any future situation
```python
if 'id' not in system and 'host_id' in system:
    system['id'] = system['host_id']
```
@renzon @rochacbruno @abalakh what do you think?

Reply:
if system is a dict we can use:

```python
for system in attrs.get('systems', []):
    system.setdefault('id', system.get('host_id'))
```

This way if system has `id` it is kept, otherwise `host_id` is looked, if not found `id` is set to `None` (but I guess the later case is not happening)

Reply:
@rochacbruno @renzon in the case of setdefault if id and host_id do not exit it will create id=None,
in that case we need to keep the dict without changes (as we need to see that error where there is no id, and if the id is None we could not understand directly that we modified that) , what do you think?

Reply:
@rochacbruno this failure is odd because is not related to my change. I have reverted it and test still failed.

Reply:
I agree if @ldjebran on this one. I have made the change and fixed the test.